src/lib/GasTank.ts

```diff
@@ -0,0 +1,182 @@
+import { Biconomy } from '@biconomy/mexa';
+
+import { SupportedChainId } from '../constants/chains';
+import {
+    BiconomyWalletClientType,
+    BuildExecTransactionType,
+    GasTankProps,
+    WebHookAttributesType,
+    ZeroWalletProviderType
+} from '../types';
+import { delay } from '../utils/global';
+import { getTransactionReceipt } from '../utils/provider';
+
+export class GasTank {
+    // public fields
+    gasTankName: string;
+    chainId: SupportedChainId;
+
+    // private fields
+    #provider: ZeroWalletProviderType;
+    #apiKey: string;
+    #fundingKey: string;
+    #biconomy = {} as any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Completely abstract out biconomy from the codebase.
We shouldn't see the word biconomy anywhere except a file called biconomyRelayer.js

BiconomyRelayer implements Relayer
GasTank instantiates BiconomyRelayer

Tomorrow if we want to use stackup we should implement StackupRelayer that extends Relayer.
GasTank will initiate StackupRelayer instead of BiconomyRelayer and change absolutely no ther code

Reply:
Got it.
I'll try to make the BaseRelayer as generic as possible.